app/models/report.rb

```diff
@@ -70,11 +70,14 @@ def links
   end
 
   def location
-    "#{reportable.name} #{reportable.class}"
+    "#{reportable.name} #{reportable_type}"
   end
 
   def sector_name
-    reportable&.sector&.name || ''
+    sector = reportable.hierarchy&.find { |geo| geo['parent_type'] == 'Sector' }
```

Comment:
I think you're on a better track here, but then it seems like you kinda make room for my really bad association chaining.

I was making every Geography layer (Country, District, Sector, Cell, Village, Facility) respond to `.sector` just so I could universally call it on any Geography. But using the hierarchy hash is MUCH cleaner. I'm going to try and stick with this first line here and handle the situations where the Report has no sector differently.